sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -3422,6 +3422,74 @@ object GpuOverrides extends Logging {
           GpuVarianceSamp(childExprs.head, !legacyStatisticalAggregate)
         }
       }),
+    expr[Percentile](
+      "Aggregation computing exact percentile",
+      ExprChecks.reductionAndGroupByAgg(
+        // The output can be a single number or array depending on whether percentiles param
+        // is a single number or an array.
+        TypeSig.DOUBLE + TypeSig.ARRAY.nested(TypeSig.DOUBLE),
+        TypeSig.DOUBLE + TypeSig.ARRAY.nested(TypeSig.DOUBLE),
+        Seq(
+          // ANSI interval types are new in Spark 3.2.0 and are not yet supported by the
+          // current GPU implementation.
+          ParamCheck("input", TypeSig.integral + TypeSig.fp, TypeSig.integral + TypeSig.fp),
+          ParamCheck("percentage",
+            TypeSig.lit(TypeEnum.DOUBLE) + TypeSig.ARRAY.nested(TypeSig.lit(TypeEnum.DOUBLE)),
+            TypeSig.DOUBLE + TypeSig.ARRAY.nested(TypeSig.DOUBLE)),
+          ParamCheck("frequency",
+            TypeSig.LONG + TypeSig.ARRAY.nested(TypeSig.LONG),
+            TypeSig.LONG + TypeSig.ARRAY.nested(TypeSig.LONG)))),
+      (c, conf, p, r) => new TypedImperativeAggExprMeta[Percentile](c, conf, p, r) {
+        override def tagAggForGpu(): Unit = {
+          // Check if the input percentage can be supported on GPU.
+          childExprs(1).wrapped match {
```

Comment:
nit: can we use extractLit instead of matching on Literal directly? It is slightly more robust.

```
extractLit(childExprs(1)) match {
  case None =>
    willNotWorkOnGpu("percentile on GPU only supports literal percentages")
  case Some(Literal(null, _)) =>
    willNotWorkOnGpu("percentile on GPU only supports non-null literal percentages")
  case (Some(Literal(a: ArrayData, _)) if a.toDoubleArray()
                .exists(percentage => percentage < 0.0 || percentage > 1.0) =>
    willNotWorkOnGpu("percentile requires the input percentages given in the range [0, 1]")
  case Some(_) => // This is fine
}
```

Reply:
Unfortunately I tried several ways but that doesn't work:
```
GpuOverrides.scala:3445: type mismatch;
 found   : com.nvidia.spark.rapids.BaseExprMeta[_$70] where type _$70
 required: org.apache.spark.sql.catalyst.expressions.Expression
```

Reply:
Done.